### PR TITLE
Deprecate Comparable

### PR DESCRIPTION
**Pull request by terrorfisch**
Deprecate comparable because it is "unpythonic" and makes comparisons hard to follow.

Three subclasses have been changed:

 - `AWG` now directly implements an `id` based hash and comparison.
 - `Waveform` is now compared and hashed based on the `__slots__` and `_duration` of the base class by default. `TransformingWaveform` overrides that because it caches results for more efficient multi channel sampling.
 - `Transformation` implementations now provide there own `__eq__` and `__hash__`.

### CODE REVIEW

**Comment by terrorfisch** [comment]
@Nomos11 Do you thiunk this is reasonable?

File: qupulse/hardware/awgs/base.py [hunk 1]
```diff
@@ -142,6 +141,13 @@ def sample_rate(self) -> float:
     def compare_key(self) -> int:
         """Comparison and hashing is based on the id of the AWG so different devices with the same properties
         are ot equal"""
+        warnings.warn("AWG.compare_key is deprecated since 0.11 and will be removed in 0.12",
+                      DeprecationWarning, stacklevel=2)
```
Comment (terrorfisch) [inline qupulse/hardware/awgs/base.py:145]: The return value us missing here

File: qupulse/program/transformation.py [hunk 2]
```diff
@@ -241,8 +277,16 @@ def get_input_channels(self, output_channels: AbstractSet[ChannelID]) -> Abstrac
     def get_output_channels(self, input_channels: AbstractSet[ChannelID]) -> AbstractSet[ChannelID]:
         return input_channels
 
+    def __eq__(self, other):
+        return isinstance(other, OffsetTransformation) and self._offsets == other._offsets
```
Comment (terrorfisch) [inline qupulse/program/transformation.py:281]: Should be the same over all Transformations with appropriate reasoning.

```
self._attribute == getattr(other, '_attribute ', None)
isinstance(other, SomeTransformation) and self._attribute == other._attribute 
```

**Comment by terrorfisch** [comment]
@Nomos11 please merge if you don't have objections

**Comment by Nomos11** [comment]
I'm not authorized to merge into master.

**Review by Nomos11 — APPROVED** [review]
